drivers/serial/uart_ns16550.c

```diff
@@ -601,7 +612,7 @@ static int uart_ns16550_fifo_fill(const struct device *dev,
 	int i;
 	k_spinlock_key_t key = k_spin_lock(&DEV_DATA(dev)->lock);
 
-	for (i = 0; (i < size) && (INBYTE(LSR(dev)) & LSR_THRE) != 0; i++) {
+	for (i = 0; (i < size) && i < DEV_DATA(dev)->fifo_size; i++) {
```

Comment:
Should this check the LSR_THRE bit once before the loop?

Reply:
The fifo_fill function can be called only from ISR after uart_irq_tx_ready() returning true. AFAIK and according to tests, the interrupt is triggered when the FIFO is empty, so IMO checking it one more time is not necessary.

Reply:
Sounds good. The ``i < fifo_size`` check should probably have enclosing parenthesis to match the existing code style.

Reply:
Done.